testing/architecture-test/src/test/java/org/gradle/architecture/test/DeclarativeDslDependencyTest.java

```diff
@@ -33,4 +37,37 @@ public class NoKotlinStdlibBinaryDependencyTest {
             .resideInAPackage("org.gradle.declarative.dsl.schema").or()
             .resideInAPackage("org.gradle.declarative.dsl.evaluation")
             .should().onlyDependOnClassesThat().haveNameNotMatching("kotlin\\.(?!Metadata$).*");
+
+    @ArchTest
+    public static final ArchRule evaluator_classes_do_not_depend_on_generic_gradle_code =
+        // The `declarative-dsl-evaluator` is meant to be used outside the context of Gradle.
+        // We check that it does not depend on Gradle API/internals.
```

Comment:
💭 You could add a `.because()` call to the end of the chain and move this comment there so that this explanation would be immediately present in the error message when this test fails.

Reply:
Good point, thank you.